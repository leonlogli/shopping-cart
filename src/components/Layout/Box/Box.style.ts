import styled, { css } from 'styled-components'
import Box from '@material-ui/core/Box'

import { BoxProps } from '.'

export const StyledBox = styled(Box)<BoxProps>`
  ${({ fullWidth }) =>
    fullWidth &&
    css`
      width: 100%;
    `}
`

export default StyledBox
