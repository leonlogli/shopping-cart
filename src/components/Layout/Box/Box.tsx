import React, { forwardRef } from 'react'
import { BoxProps as BaseBoxProps } from '@material-ui/core/Box'

import StyledBox from './Box.style'

export type BoxProps = BaseBoxProps & {
  /**
   * If `true`, the box will take up the full width of its container.
   * @default false
   */
  fullWidth?: boolean
}

/**
 * The Box component serves as a wrapper component for most of the CSS utility needs.
 */
const Box = forwardRef<HTMLDivElement, BoxProps>((props, ref) => {
  const { children, ...other } = props
  // Workaround to pass ref without having type checking issue
  const refProp = { ref }

  return (
    <StyledBox {...refProp} {...other}>
      {children}
    </StyledBox>
  )
})

export { Box }
export default Box
