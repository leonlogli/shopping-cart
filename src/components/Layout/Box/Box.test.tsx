import React from 'react'
import { mount } from 'enzyme'
import Box from './index'

describe('<Box />', () => {
  const wrapper = mount(<Box>text</Box>)

  it('should have the correct children', () => {
    expect(wrapper.props().children).toBe('text')
  })

  it('should have the default component', () => {
    expect(wrapper.find('div').length).toEqual(1)
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
