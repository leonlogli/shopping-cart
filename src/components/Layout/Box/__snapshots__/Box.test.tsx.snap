// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should match a snapshot 1`] = `
<ForwardRef>
  <Styled(Styled(MuiBox))>
    <Styled(MuiBox)
      className="njkSP"
    >
      <div
        className="MuiBox-root MuiBox-root-1 njkSP"
      >
        text
      </div>
    </Styled(MuiBox)>
  </Styled(Styled(MuiBox))>
</ForwardRef>
`;
